### ISSUE TITLE
BackendConfig support for user-defined request headers

### ISSUE THREAD

**Issue by acasademont** [kubernetes/ingress-gce#566]
Would be great if BackendConfig had support for user-defined request headers as per https://cloud.google.com/load-balancing/docs/backend-service#user-defined-request-headers

**Comment by pondohva** [comment]
I'll be happy to implement this feature. I'd appreciate if you give me some references to similar feature.

**Comment by pondohva** [comment]
Never mind, I think I got the point.

**Comment by caquino** [comment]
I'm also looking for this feature to enable custom geolocation headers on the backends.

Is this something that is still being worked on?

**Comment by rramkumar1** [comment]
We had a PR to implement this (#572) but it seems the PR has been abandoned. Please feel free to pick it up.

**Comment by mcfedr** [comment]
Fixed in #795

**Comment by thuandt** [comment]
@mcfedr seem Google is not yet rolling this feature.

I tried with this configuration in GKE cluster v1.13.x but didn't work

```apiVersion: cloud.google.com/v1beta1
kind: BackendConfig
metadata:
  namespace: default
  name: my-cdn
spec:
  # https://cloud.google.com/load-balancing/docs/user-defined-request-headers
  # https://blog.doit-intl.com/farewell-maxmind-geolocating-requests-with-google-load-balancer-for-free-7938f5adbef0
  # https://github.com/kubernetes/ingress-gce/pull/795/files
  customRequestHeaders:
    - "X-Client-Geo-Location:{client_region},{client_region_subdivision},{client_city}"
    - "X-Client-Geo-Region:{client_region_subdivision}"
    - "X-Client-Geo-LatLong:{client_city_lat_long}"
    - "CloudFront-Viewer-Country:{client_region}"
  cdn:
    enabled: true
    cachePolicy:
      includeHost: true
      includeProtocol: true
      includeQueryString: false```

**Comment by rramkumar1** [comment]
@thuandt This has not been released yet on GKE. It will likely be coming in our 1.7 release. Stay tuned...

**Comment by thuandt** [comment]
@rramkumar1 Ok. Noted

I tried with above configuration in GKE (us-west1), GCP has been trigger to add new custom headers in only one backend services instead both of them so it's not worked as expected.

btw, thanks for your support :)

**Comment by mitulpanch** [comment]
any update on this?

**Comment by rramkumar1** [comment]
@mitulpanch This is now shipped as part of our 1.7 release.
